clpy/core/core.pyx

```diff
@@ -3074,23 +3074,26 @@ cpdef _scatter_op_single(ndarray a, ndarray indices, v,
         (1,) * len(lshape) + indices_shape + (1,) * len(rshape))
     indices = broadcast_to(indices, v_shape)
 
-    raise NotImplementedError("clpy does not support this")
-#    if op == 'update':
-#        _scatter_update_kernel(
-#            v, indices, cdim, rdim, adim, a.reduced_view())
-#    elif op == 'add':
-#        # There is constraints on types because atomicAdd() in CUDA 7.5
-#        # only supports int32, uint32, uint64, and float32.
-#        if not issubclass(v.dtype.type,
-#                          (numpy.int32, numpy.float32,
-#                           numpy.uint32, numpy.uint64, numpy.ulonglong)):
-#            raise TypeError(
-#                'scatter_add only supports int32, float32, uint32, uint64 as '
-#                'data type')
-#        _scatter_add_kernel(
-#            v, indices, cdim, rdim, adim, a.reduced_view())
-#    else:
-#        raise ValueError('provided op is not supported')
+    if op == 'update':
+        # _scatter_update_kernel(
+        #     v, indices, cdim, rdim, adim, a.reduced_view())
+        raise NotImplementedError("clpy does not support this")
```

Comment:
Is there any test for this path? Which is?

Reply:
No tests is contained in this PR.
This PR supports only `scatter_add` for float32, and the path, which is called on `ndarray.__setitem__` , behave as same as current ClPy.
Of course, it should be supported in the future (in another PR).

Reply:
Same as https://github.com/fixstars/clpy/pull/66#discussion_r215852751